File: src/main/java/qmech/lib/objects/CreativeTabBase.java
package qmech.lib.objects;

import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.Item;
import qmech.mod.Reference;

public class CreativeTabBase extends CreativeTabs {

    public Item icon;

    public CreativeTabBase(String tabLabel, Item icon) {
        super(CreativeTabs.getNextID(), tabLabel);
        this.icon = icon;
    }

    public static CreativeTabBase add(String label, Item iconIndex) {
        String name = Reference.MOD_NAME + " | " + label;
        return new CreativeTabBase(name, iconIndex);
    }

    public Item getTabIconItem() {
        return this.icon;
    }

}

File: src/main/scala/qmech/lib/objects/CreativeTabBase.scala
package qmech.lib.objects

import net.minecraft.item.Item
import net.minecraft.creativetab.CreativeTabs

/**
 * Created by anshuman on 26-05-2014.
 */
class CreativeTabBase(val tabLabel: String, val icon: Item) extends CreativeTabs(CreativeTabs.getNextID, tabLabel) {
  override def getTabIconItem: Item = icon
}
